module/library/surveys/helper.js

```diff
@@ -0,0 +1,125 @@
+*/
+module.exports = class SurveyHelper {
+
+    /**
+      * List of library solution
+      * @method
+      * @name list
+      * @param {String} search - Search Data.
+      * @param {Number} limit - limitting value.
+      * @param {Number} pageNo
+      * @returns {Object} returns list of survey solutions
+     */
+
+    static list(search,limit,pageNo,userId,token) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                let librarySolution = 
+                await solutionsHelper.templates( 
+                  messageConstants.common.SURVEY,
+                  search, 
+                  limit, 
+                  pageNo, 
+                  userId,
+                  token   
+                );
+
+                return resolve({
+                  success: true,
+                  message : messageConstants.apiResponses.SURVEY_SOLUTIONS_LIST_FETCHED,
+                  data : librarySolution
+                });
+
+            } catch (error) {
+                return resolve({
+                    success: false,
+                    message: error.message,
+                    data: false
+                });
+            }
+        })
+    }
+
+    /**
+      * Survey solution details information
+      * @method
+      * @name details
+      * @param {String} templateId - Template id.
+      * @returns {Object} returns creator,about and questions of survey solutions.
+     */
+
+    static details(templateId) {
+        return new Promise(async (resolve, reject) => {
+            try {
+                
+                let solutionDetails = 
+                await solutionsHelper.details(
+                  templateId
+                );
+
+                let criteriaIds = gen.utils.getCriteriaIds(solutionDetails.themes);
+
+                let questionDetails = 
+                await criteriaQuestionsHelper.details(
+                    criteriaIds,
+                    {
+                        $project : {
+                            "ecm" : "$evidences.code",
+                            "question" : "$evidences.sections.questions.question"
+                          }
+                    }
+                 );
+
+                let questions = [];
+
+                if( questionDetails.length > 0 ) {
+                  questions = questionDetails.map(questionData=>{
+                    return questionData.question[0]
+                  });
+
+                  questions = [...new Set(questions)]
+                }
+
+                let result = {
+                  isReusable: false,
+                  allowImport: false,
+                  name : solutionDetails.name,
+                  creator : solutionDetails.creator ? solutionDetails.creator : "",
+                  description : solutionDetails.description,
+                  entityType : solutionDetails.entityType,
+                  linkTitle : solutionDetails.linkTitle ? solutionDetails.linkTitle : "",
```

Comment:
@deepakr123 - Figure out who has done this code, who is using the code, do we still need it and what should be the value here for survey template details

Reply:
resolved